Mono.Debugging/Mono.Debugging.Client/BreakpointStore.cs

```diff
@@ -326,12 +351,12 @@ public ReadOnlyCollection<Breakpoint> GetBreakpointsAtFileLine (string filename,
 
 		public IEnumerator GetEnumerator ()
 		{
-			return breakpoints.GetEnumerator ();
+			return InternalGetBreakpoints ().ToList().GetEnumerator ();
```

Comment:
Why ToList? It is not necessary. Same below.

Reply:
`breakpoints` is an ImmutableArray, so by nature it can't change. It is totally safe to return an enumerator for it, there is no need to create a temporary copy with ToList()

Reply:
Because of 

```
/Users/gregm/Projects/vsm/a/monodevelop/main/external/debugger-libs/Mono.Debugging/Mono.Debugging.Client/BreakpointStore.cs(11,11): Error CS0029: Cannot implicitly convert type 'System.Collections.Immutable.ImmutableArray<Mono.Debugging.Client.BreakEvent>.Enumerator' to 'System.Collections.IEnumerator' (CS0029) (Mono.Debugging)
```

Reply:
Even though GetEnumerator() is not exposed as a public method, ImmutableArray does implement IEnumerator, so  you can do: `((IEnumerable)breakpoints).GetEnumerator()`

Reply:
I don't think it does implement IEnumerator. It does implement IEnumerable but I cannot see it implementing IEnumerator.

Reply:
I mean that it implements IEnumerable, sorry, so you can get an enumerator from it.

Reply:
🤦‍♂